sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuUserDefinedFunction.scala

```diff
@@ -75,6 +75,17 @@ object GpuUserDefinedFunction {
   // UDFs can support all types except UDT which does not have a clear columnar representation.
   val udfTypeSig: TypeSig = (TypeSig.commonCudfTypes + TypeSig.DECIMAL_64 + TypeSig.NULL +
       TypeSig.BINARY + TypeSig.CALENDAR + TypeSig.ARRAY + TypeSig.MAP + TypeSig.STRUCT).nested()
+
+  /** Detect whether the Java assertion is going to be enabled for the `clsName`. */
+  def isClassAssertionEnabled(clsName: String): Boolean = try {
+    Class.forName(clsName).desiredAssertionStatus()
+  } catch {
+    case _: ClassNotFoundException => false
+  }
+
+  /** (This will be initialized once per process) */
+  lazy val hostColumnAssertionEnabled: Boolean =
+    isClassAssertionEnabled("ai.rapids.cudf.HostColumnVectorCore")
```

Comment:
`HostColumnVectorCore` is public. Why not just do the following instead?

`classOf[HostColumnVectorCore].desiredAssertionStatus()`

Reply:
Good suggestion, updated